sdk/eventgrid/azure-eventgrid/samples/async_samples/eventgrid_client_samples/sample_binary_mode_async.py

```diff
@@ -0,0 +1,49 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import asyncio
+import json
+from azure.core.credentials import AzureKeyCredential
+from azure.eventgrid.aio import EventGridClient
+from azure.eventgrid.models import *
+from azure.core.messaging import CloudEvent
+from azure.core.exceptions import HttpResponseError
+
+
+EVENTGRID_KEY: str = os.environ["EVENTGRID_KEY"]
+EVENTGRID_ENDPOINT: str = os.environ["EVENTGRID_ENDPOINT"]
+TOPIC_NAME: str = os.environ["EVENTGRID_TOPIC_NAME"]
+EVENT_SUBSCRIPTION_NAME: str = os.environ["EVENTGRID_EVENT_SUBSCRIPTION_NAME"]
+
+# Create a client
+client = EventGridClient(EVENTGRID_ENDPOINT, AzureKeyCredential(EVENTGRID_KEY))
+
+
+async def run():
+    async with client:
+        # Publish a CloudEvent
+        try:
+            cloud_event_dict = {"data":b"HI", "source":"https://example.com", "type":"example", "datacontenttype":"text/plain"}
+            await client.publish_cloud_events(topic_name=TOPIC_NAME, body=cloud_event_dict)
+
+            cloud_event = CloudEvent(data=json.dumps({"hello":"data"}).encode("utf-8"), source="https://example.com", type="example", datacontenttype="application/json")
```

Comment:
non-blocking, but might be helpful to add a comment to say what you're showing for each cloudEvent that you construct here. something like/better than:
line 29
```
# publish cloud event dict
...
# binary encode dict data, then publish in binary_mode
...
```